### PR TITLE
Improvements in the usability of the debug console

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Improvement.
#### Summary of Changes

This PR adds a fixed position and ability to drag the debug console.

Now the page is always the same but with a debug console that is fixed in a position. No need to scroll down to the bottom of the page to check the debug results.
#### Testing Instructions
1. Use latest staging.
2. Enable the debug console.
3. Check the debug console is always on the bottom of the page (you have to scroll down always).
4. Apply patch.
5. Check the debug console is always fixed on top right of the screen and opens on clear.
6. Check you can drag the console to another location if you need (with the drag icon).
7. On a long page, scroll down and see if the debug console is always there.

image
#### Observations

Please note the js files added in this PR are the `jquery ui` `draggable` events.

Improvements suggestions are welcomed.

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
ups, notice some strange things. will need more work. don't test yet.

**Comment by brianteeman** [comment]
I know where you are going and I love the idea. One thing I would add would be a collapse/expand all link in the title bar<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@brianteeman can you check? na give feedback.

**Comment by MATsxm** [comment]
That is **REALLY great** :+1: 

Just note that if you open the console, it is then impossible to reach the icon to drag it until you close all the tabs.

screen shot 2016-04-14 at 10 55 45

Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on ba9821f87c6935b339c608dc9b0b112c5349abbc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by infograf768** [comment]
Unsuccessful here:
Some debug content is clipped and no way to access to it, even on big screens (not speaking of modals).
Example: before patch I can go to bottom of the Language Files Loaded :


screen shot 2016-04-14 at 18 15 15


After patch I can't (missing 10 of them):

screen shot 2016-04-14 at 18 11 43

**Comment by infograf768** [comment]
I have tested this item :red_circle: unsuccessfully on ba9821f87c6935b339c608dc9b0b112c5349abbc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
humm seems we need to have a small max-height for the content area...

**Comment by brianteeman** [comment]
Two issues 
1.no matter where you move it to or what you have expanded - it moves back to being minimised and on the top right on a page load
2. As @infograf768 says you miss the very bottom of the output - not just language strings the same can be seen with queries<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
reggarding "2. As @infograf768 says you miss the very bottom of the output - not just language strings the same can be seen with queries"

i think now it will show.

**Comment by infograf768** [comment]
Will test tomorrow,  I suggest to also test in modal views.

**Comment by andrepereiradasilva** [comment]
yeah, more work is needed i guess.

**Comment by brianteeman** [comment]
The height issue is fixed

New problems (or ones I just didnt spot before)

1. Profile information Time and Memory bars are larger than the container box
2. Database query explains are larger than the container box (note these only break out of the container on hover)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
the height issue is also solved in small height screens and modals since if those it uses the current (pre PR) display.

**Comment by andrepereiradasilva** [comment]
@brianteeman can you attach a screenshot?

**Comment by brianteeman** [comment]
Better than that - grab some popcorn and watch the movie

http://i.tee.mn/debug2.gif<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
ok, i think those issues are solved now.

**Comment by andrepereiradasilva** [comment]
forget it still some issues. ...

**Comment by andrepereiradasilva** [comment]
@brianteeman can you check now?

image

**Comment by brianteeman** [comment]
The issue about breaking out of the box are now resolved.

Can you look at the login screen
gx5d

**Comment by andrepereiradasilva** [comment]
@brianteeman what is the issue with the login screen?

**Comment by brianteeman** [comment]
without new-debug enabled the login is about 170px from the top of the screen
with new-deug enabled its as per the screenshot

> 1.no matter where you move it to or what you have expanded - it moves back to being minimised and on the top right on a page load

Still annoys me a lot

**Comment by andrepereiradasilva** [comment]
> without new-debug enabled the login is about 170px from the top of the screen
> with new-deug enabled its as per the screenshot

actually that already happens with the current debug

**Comment by brianteeman** [comment]
Ah I see what you mean - I guess because the debug was pushed up against the login before it wasnt as obvious. Not a big deal

**Comment by andrepereiradasilva** [comment]
i think it's an html error or something that puts the browser in quirks rendering mode or something.

I will check that also when i have time.

**Comment by andrepereiradasilva** [comment]
@brianteeman i just discovered that the login form not centering has not to do with the system debug plugin, but with the debug global configuration setting.

You can check that by:
1. Enabling debug in system config
2. Disabling the system debug plugin
3. Logoff (or use other anonymous browser tab)
4. See that there is no debug console and yet the login form screen is not centered.

So i will not solve that is this PR.

**Comment by andrepereiradasilva** [comment]
@brianteeman, just discovered why the login form goes to top on debug mode.

See https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/isis/login.php#L88-L98

**Comment by brianteeman** [comment]
Which can be removed then with your PR ;)

On 15 April 2016 at 12:37, andrepereiradasilva <notifications@github.com>
wrote:

> @brianteeman <https://github.com/brianteeman>, just discovered why the
> login form goes to top on debug mode.
>
> See
> https://github.com/joomla/joomla-cms/blob/staging/administrator/templates/isis/login.php#L88-L98
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/9910#issuecomment-210427642>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by andrepereiradasilva** [comment]
... ok

**Comment by designbengel** [comment]
there are still issues with breaking the box
https://www.dropbox.com/s/8yivod42j7o7gm2/Screenshot%202016-04-15%2014.44.37.png?dl=0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@designbengel thanks. what browser did you used?

**Comment by designbengel** [comment]
... Sorry, sure! I forgot to add that information. It´s Chrome Version 49.0.2623.112 (64-bit) 
and: i accidentally moved out the box outside the screen and couldn´t move it back again.

**Comment by andrepereiradasilva** [comment]
> ... Sorry, sure! I forgot to add that information. It´s Chrome Version 49.0.2623.112 (64-bit) 

Ok will check.

>  i accidentally moved out the box outside the screen and couldn´t move it back again.

refresh the page :smile: eheh

**Comment by designbengel** [comment]
I tested another patch with opera Browser with yours still applied and the box looks like this in Opera:
https://www.dropbox.com/s/jyujbi1qgpw9i9v/Screenshot%202016-04-15%2014.54.05.png?dl=0

**Comment by andrepereiradasilva** [comment]
that's really strange since opera rendering machine is the same as chrome.

**Comment by cheiff** [comment]
I have tested this item :white_check_mark: successfully on 5651766267ac0f9e5455193d90eab9993a228fce<br /><br />I have tested this successfully in Firefox 45 and Chrome 50.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
mobile issues on login screen solved.

now i need to check the @designbengel visual problem and after it should be ready for testing.

**Comment by andrepereiradasilva** [comment]
@designbengel the opera problem seems to be because of your screen size.

The static floating debug console only apperas when screen size `@media (max-height: 650px), (max-width: 1260px)`. in other order width higher than 1260px and height higher than 650px.

The chrome profile bars problem i can't reproduce it. Are you using chrome for windows?

**Comment by MATsxm** [comment]
I have tested this item :white_check_mark: successfully on bd398b2302c73635487b471095a05ee9d9ccc8c3<br /><br />Tested (zoom 100%) on **Chrome** 50.0.2661.75 m (+Yandex-Chromium) / **FF** 45.0.2 / **Edge** 25.105 and everything looks fine here, even issues previously reported.
Thanks for this great one<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by brianteeman** [comment]
I still have an issue with the time and memory bars breaking out of the window 
This screenshot shows in order from left to right
Safari Firefox Chrome all on osx
screen shot 2016-04-22 at 09 50 17<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
the thing is i don't have a mac to check that one...
anyone can point what is worng there on mac?

**Comment by brianteeman** [comment]
I will try and spend some time tomorrow to inviestigate<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
@brianteeman did you manage to get time to check the mac issue?

**Comment by brianteeman** [comment]
Sorry not yet - but i have not forgotten

File: media/cms/css/debug.css [hunk 1]
```diff
@@ -49,6 +66,10 @@ div#system-debug {
 
 #system-debug div.dbg-container {
 	padding: 10px;
+	overflow-y: scroll;
+	max-height: 350px;
+	min-width: 80em;
```
Comment (brianteeman) [inline media/cms/css/debug.css]: Changing this to about 78em fixed the issue with the countainer being bigger than the frame and the bars sticking out

Comment (andrepereiradasilva) [inline media/cms/css/debug.css]: thanks will check that.

Comment (brianteeman) [inline media/cms/css/debug.css]: sooner the better or it wont make it into 3.6 ;)

**Comment by andrepereiradasilva** [comment]
changed from 80em to 78em as suggested.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5237c8f57db8c3f103a69e54590b45df256b3f5e<br /><br />Perfect thanks

One more quick test please<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by andrepereiradasilva** [comment]
there a js error when you have system debug diabled and system plugin enabled. 
don't test yet.

**Comment by andrepereiradasilva** [comment]
solved

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 24723ea58dc048d95633f6e6790e158efb071ce9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by conconnl** [comment]
I have tested this item :white_check_mark: successfully on 24723ea58dc048d95633f6e6790e158efb071ce9<br /><br />A great improvement indeed.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by brianteeman** [comment]
Finally RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by infograf768** [comment]
Conflicts.

**Comment by andrepereiradasilva** [comment]
conflicts fixed

**Comment by ggppdk** [comment]
I have tested this item :white_check_mark: successfully on 7bc5fbc591e73e2756d17098bb72fc6a168d915c<br /><br />Just a comment, 
about dragging, i would prefer that the full header "Joomla Debug Console" was the drag handle

just asking why use:
handle: ".header .drag",

and not ?:
handle: ".header",<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/9910">issues.joomla.org/joomla-cms/9910</a>.</sub>

**Comment by wilsonge** [comment]
OK So I'm not convinced that this is actually useful. I know scrolling down to the bottom of the page is a pain. But my opinion is that having to drag this around into exactly the correct position is going to be an even bigger pain. On top of that it's still not trivial to use. For example testing now I moved it into roughly the center of the screen. Opened the session tab and then found that it expanded off the edge of the page and i couldn't drag it, I then had to close the console, reposition the modal and reopen it again. But even aside from that (bug?) is it really that much easier than what we have now?

**Comment by andrepereiradasilva** [comment]
Actually @wilsonge i'm not convinced too now.

I been thinking of closing this PR for sometime.
I now think the debug panel should be like a sidepanel menu or something, but i lack of css knowledge to do that. Somethink like this http://www.w3schools.com/howto/howto_js_sidenav.asp

So i will close this. Sorry all testers but i don't want to pursue this anymore.

**Comment by gorgonz** [comment]
could someone remove milestone 3.7 or something appropriate, because i try to create a list of new feature in 3.7 and it shouldn't be listed at the moment ;-)

**Comment by wilsonge** [comment]
Removed - thanks for the prompt

**Pull request closed without merging**